### ISSUE TITLE
3.3.0 segmentation fault 11 on OSX

### ISSUE THREAD

**Issue by bestander** [sass/node-sass#1107]
Node quits with "segmentation fault 11" when node-sass is required.
I am using gulp-sass that depends on node-sass.

Mac OS X 10.10.5 
node -v
v2.3.0
npm -v
2.12.1

**Comment by bestander** [comment]
Reverting to 3.2.0 fixes the bug

**Comment by eddiemoore** [comment]
:+1: I'm having the exact same issue
```shell
node -v
# v2.5.0
npm -v
# 2.13.2
```

I'm using it through gulp-sass though, so can't really revert.

**Comment by xzyfer** [comment]
My guess is that you need to do a delete your `node_modules` and do a clean install. Is sounds like your system has kept the v3.2.0 rather than downloading the updated v3.3.0 binary.

**Comment by bestander** [comment]
Nope, 3.3.0 is definitely causing the problem on my PC. 
Reverting 3.3.0 with 3.2.0 manually fixes the problem.

**Comment by xzyfer** [comment]
@bestander please follow our [troubleshooting guide](https://github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md#linuxosx) and create a gist with all the output.

We're unable to reproduce this issue.

**Comment by yizhan** [comment]
I deleted my node_modules, and did a clean install. It shows segmentation fault 11 now.

**Comment by xzyfer** [comment]
If someone could produce a scss sample that produce that the segfault that would greatly help.

**Comment by bestander** [comment]
Sure will do.  
I am using gulp-sass, so maybe the reason is that latest gulp-sass is not compatible with node-sass 3.3.0

**Comment by enriquecaballero** [comment]
Same here. Downgrading to 3.2.0 fixes the Segmentation Fault: 11.

**Comment by bestander** [comment]
Hm, in an isolated case it does not throw the error:
```
var sass = require('gulp-sass');
var gulp = require('gulp');

gulp.task('test', function () {

});
```

But in our production system with a large gulp script just having 
```
var sass = require('gulp-sass');
```
Throws node with error.

```
npm -v
2.12.1
node -v
v2.3.0
node -p process.versions
{ http_parser: '2.5.0',
  node: '2.3.0',
  v8: '4.2.77.20',
  uv: '1.6.1',
  zlib: '1.2.8',
  ares: '1.10.1-DEV',
  modules: '44',
  openssl: '1.0.2c' }
node -p process.platform
darwin
node -p process.arch
x64
```

I'll try to investigate more

**Comment by xzyfer** [comment]
It's not related to gulp-sass as I am currently using that without issue.

**Comment by enriquecaballero** [comment]
@xzyfer I'm using the sass-loader for Webpack, that uses node-sass, and 3.3.0 definitely crashes Node with Segmentation Fault: 11. Downgraded to 3.2.0 and voila, it works.

**Comment by xzyfer** [comment]
@eddiemoore @enriquecaballero @yizhan 

Are you using node-sass via Gulp / Grunt, and which?

Please run the following commands

```
npm -v
node -v
node -p process.versions
node -p process.platform
node -p process.arch
```

**Comment by xzyfer** [comment]
By pure fluke I was able to reproduce this in a `gulp watch` task. The stacktrace for segfault is

```
PID 42935 received SIGSEGV for address: 0x46
0   segfault-handler.node               0x00000001013fa90f _ZL16segfault_handleriP9__siginfoPv + 287
1   libsystem_platform.dylib            0x00007fff8e80a5aa _sigtramp + 26
2   ???                                 0x0000000103300450 0x0 + 4348445776
3   fse.node                            0x00000001036ee616 _ZN3Nan3imp10GetWrapperIPFvRKNS_20FunctionCallbackInfoIN2v85ValueEEEENS0_15FunctionWrapperEEEPT0_T_ + 142
4   fse.node                            0x00000001036eda40 _ZN3fse8FSEvents10InitializeEN2v86HandleINS1_6ObjectEEE + 194
5   node                                0x0000000100601047 _ZN4node6DLOpenERKN2v820FunctionCallbackInfoINS0_5ValueEEE + 725
6   node                                0x00000001001749ee _ZN2v88internal25FunctionCallbackArguments4CallEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEE + 158
7   node                                0x000000010019c97b _ZN2v88internalL19HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateERNS0_12_GLOBAL__N_116BuiltinArgumentsILNS0_21BuiltinExtraArgumentsE1EEE + 811
8   node                                0x000000010019f1cd _ZN2v88internalL21Builtin_HandleApiCallEiPPNS0_6ObjectEPNS0_7IsolateE + 61
9   ???                                 0x00001395f2e060bb 0x0 + 21534745845947
```

The culprit appears to `fse.node` which is part of the [fsevents modules](https://www.npmjs.com/package/fsevents). This module is used in [chokidar](https://www.npmjs.com/package/chokidar) which in turn is use by many watchers like gulp and webpack.

I'm still digging into why node-sass is triggering this segfaulting behaviour in fsevents.

**Comment by xzyfer** [comment]
Some suspect things I've noticed with regards to fsevents.

- segmentation fault bug was fixed in the [0.3.7 release](https://github.com/strongloop/fsevents/releases/tag/v0.3.7) 26 days ago.
- they also converted to `nan@2` [0.3.8 release](https://github.com/strongloop/fsevents/releases/tag/v0.3.8) 19 days ago.

**Comment by eddiemoore** [comment]
Incase you still need it
```shell
node -v
# v2.5.0
npm -v
# 2.13.2
node -p process.versions
{ http_parser: '2.5.0',
  node: '2.5.0',
  v8: '4.2.77.21',
  uv: '1.6.1',
  zlib: '1.2.8',
  ares: '1.10.1-DEV',
  modules: '44',
  openssl: '1.0.2d' }
node -p process.platform
# darwin
node -p process.arch
# x64

**Comment by xzyfer** [comment]
@eddiemoore are you experiencing this through a watcher or in a singe gulp run?

**Comment by eddiemoore** [comment]
both. During a build and watch.
```js
gulp.task('styles', () => {
  return gulp.src('src/styles/**/*.scss')
    .pipe($.sass().on('error', $.sass.logError))
    .pipe(gulp.dest('dist/styles'))
})
```
Using gulp-load-plugins in case you are wondering what the $ is.

**Comment by xzyfer** [comment]
I've done some more digging and I believe this to be a bug in `fsevents@0.3.8`. By locking to `fsevents@0.3.7` I'm no longer able to reproduce this issue.

~~~~~

@enriquecaballero @eddiemoore @bestander can you please try the following:

Add this line as the first dependency on your `package.json`

```
  "dependencies": {
    "fsevents": "0.3.7"
```

Then delete your `node_modules` and do a fresh `npm install`.

**Comment by eddiemoore** [comment]
@xzyfer That solves the issue for now. Just hope this will be fixed soon without having to do this.

**Comment by xzyfer** [comment]
Thanks @eddiemoore. That just helps confirm fsevents is the culprit. I've commented on their related issue https://github.com/paulmillr/chokidar/issues/219

**Comment by vinkla** [comment]
Reverting to `3.2.0` from `3.3.0` solved this for me as well.

**Comment by eddiemoore** [comment]
Only thing is CodeShip CI can't install it.........
```
npm ERR! node v2.5.0
npm ERR! npm  v2.13.2
npm ERR! code EBADPLATFORM

npm ERR! notsup Unsupported
npm ERR! notsup Not compatible with your operating system or architecture: fsevents@0.3.7
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
```

**Comment by xzyfer** [comment]
@eddiemoore make it an optionalDependency instead

```
  "optionalDependencies": {
    "fsevents": "0.3.7"
  },
```

**Comment by eddiemoore** [comment]
:+1:

**Comment by mucahit** [comment]
Hıaaa 👊

**Comment by xzyfer** [comment]
I've pushed out v3.3.1 which has a workaround for this issue until it's fixed for good.

**Comment by bvleur** [comment]
I still get this segmentation fault with node-sass@3.3.1

**Comment by xzyfer** [comment]
You might have to do a fresh npm to delete the old fsevent installations.

If that doesn't work please post the results of an `npm ls`

**Comment by noKid** [comment]
I also still have the problem. I'm using gulp-sass which uses node-sass.
Here's the part of 'npm ls' relative to it.

├─┬ gulp-sass@2.0.4
│ ├─┬ node-sass@3.3.1
│ │ ├── async-foreach@0.1.3
│ │ ├─┬ chalk@1.1.1
│ │ │ ├── ansi-styles@2.1.0
│ │ │ ├── escape-string-regexp@1.0.3
│ │ │ ├─┬ has-ansi@2.0.0
│ │ │ │ └── ansi-regex@2.0.0
│ │ │ ├─┬ strip-ansi@3.0.0
│ │ │ │ └── ansi-regex@2.0.0
│ │ │ └── supports-color@2.0.0
│ │ ├─┬ fsevents@0.3.7

**Comment by xzyfer** [comment]
@noKid could you please provide the output of the following commands in a gist.

```
npm -v
node -v
node -p process.versions
node -p process.platform
node -p process.arch
npm ls
```

**Comment by xzyfer** [comment]
Everyone feel free to join our slack channel so we work together to figure this out without creating too much noise

https://libsass-slack.herokuapp.com/

**Comment by noKid** [comment]
[Here's the gist](https://gist.github.com/noKid/a8304fb2c6180673d9e2)

**Comment by xzyfer** [comment]
@noKid it looks like you're still getting `fsevents@0.3.8` installed by karma and karma-browserify. That's issue. I expected the v3.3.1 would work around this. For now running `npm dedupe` should help.

**Comment by xzyfer** [comment]
I've pulled the v3.3.x releases until we can get the bottom of this issue.

```
npm install node-sass
```

Will now install v3.2.0 once again. Thank you to everyone who helped out.

**Comment by iamstarkov** [comment]
v3.3.1 doesnt help on os x

**Comment by saper** [comment]
@xzyfer any recommendation how to get connected to some MacOS X instance to check this?

**Comment by xzyfer** [comment]
@saper I don't know of a way to get remote access to an OSX machine, sorry.

**Comment by xzyfer** [comment]
@saper libsass runs against osx in CI. Maybe you could hack together something to exploit that in PR?

**Comment by lomteslie** [comment]
Today I started getting the segmentation fault 11 issue. However, I'm running node-sass 3.2.0 with sass-loader for webpack. Based on this conversation it doesn't seem like that's possible. Am I losing my mind?

**Comment by vinkla** [comment]
@lomteslie Please see https://github.com/sass/node-sass/issues/1107#issuecomment-134902835

**Comment by lomteslie** [comment]
@vinkla Thank you – That fixed it after reinstalling node modules.

**Comment by xzyfer** [comment]
Yes unfortunately the issue isn't specific to node-sass but the underlying 3rd party library used for compiling native extensions like watchers used by grunt, gulp, webpack and karma. It just so happens node-sass manages to trigger this issue. We're working with the 3rd party library to get this resolved.

**Comment by saper** [comment]
This should now be fixed to the extent possible via  #1110 and #1113

**Comment by xzyfer** [comment]
`node-sass@v3.3.2` has been released - https://github.com/sass/node-sass/releases/tag/v3.3.2

**Comment by sirrodgepodge** [comment]
Hey all, just wanted to say that in my experience this can sometimes be due to particular sass bugs and sometimes only cause seg faults in one OS (node-sass does different things dep on OS), one that's caused seg faul n Linuxt:
-using interpolation without variable, e.g. (this is kinda a user error in that #{} interpolation isn't necessary):

.example {
  width: #{100%/3}
}

switched to:

.example {
  width: 100%/3
}

fixes

Happened one other time when I was using an undeclared mixin

**Comment by mgreter** [comment]
@sirrodgepodge segfaults are pretty much always a bug in libsass and if someone would have reported it as an issue, libsass team would have had a chance to fix it already ...

**Comment by xzyfer** [comment]
@sirrodgepodge this has been fixed in libsass and will be in the next node-sass version.

**Comment by spezticle** [comment]
I was experiencing this issue. Initially, I installed encfs with homebrew while my OS x version was Yosemite.
brew install homebrew/fuse/encfs

Everything worked fine until I upgraded to El Capitan, (begrudgingly). Then I experienced the segmentation fault 11. I uninstalled encfs:
brew uninstall homebrew/fuse/encfs --force encfs

and then reinstalled:
brew install homebrew/fuse/encfs

and everything worked fine again.
